src/librustc_errors/emitter.rs

```diff
@@ -721,7 +736,10 @@ impl EmitterWriter {
             draw_note_separator(&mut buffer, 0, max_line_num_len + 1);
             buffer.append(0, &level.to_string(), Style::HeaderMsg);
             buffer.append(0, ": ", Style::NoStyle);
-            buffer.append(0, msg, Style::NoStyle);
+
+            // The extra 3 ` ` is the padding that's always needed to align to the `note: `.
+            let message = self.msg_with_padding(msg, max_line_num_len + "note: ".len() + 3);
```

Comment:
[Added this documentation to the code](https://github.com/rust-lang/rust/pull/38916/commits/04e4a60b45ec4debd20be2327cb6859271502c05).